### PR TITLE
[#433] switch from gcs storage to hcapi hosted data

### PR DESCRIPTION
**Pull request by pavertomato**
based on https://github.com/GoogleCloudPlatform/healthcare/pull/423

- switch from gcs storage to hcapi hosted data - **done**
- remove redundant TCIA api code - **done**
- `gcloud beta healthcare dicom-stores export` instead of api call - **done**
- should be exported only train images that label breast density 2 or 3, avoid filter blacklisted dicoms - actually filtering already performed by https://github.com/GoogleCloudPlatform/healthcare/blob/master/imaging/ml/ml_codelab/scripts/tcia_utils.py#L77 and https://github.com/GoogleCloudPlatform/healthcare/blob/master/imaging/ml/ml_codelab/scripts/tcia_utils.py#L81

not yet resolved: access to https://healthcare.googleapis.com/v1beta1/projects/chc-tcia/locations/us-central1/datasets/cbis-ddsm/dicomStores/cbis-ddsm seems to be managed by CHC-API team https://github.com/GoogleCloudPlatform/healthcare/issues/433#issuecomment-576373726

### CODE REVIEW

File: imaging/ml/ml_codelab/scripts/preprocess/preprocess.py [hunk 1]
```diff
@@ -195,7 +195,7 @@ def _get_study_uid_to_image_path_map(input_path):
   study_uid_to_file_paths = {}
   for path in path_list:
     split_path = path.split('/')
-    study_uid_to_file_paths[split_path[3]] = path
```
Comment (pavertomato) [inline imaging/ml/ml_codelab/scripts/preprocess/preprocess.py]: this part was accidentally reverted during previous review

File: imaging/ml/ml_codelab/breast_density_cloud_ml.ipynb [hunk 2]
```diff
@@ -505,7 +320,7 @@
     "%%bash -s {location} {bottleneck_bucket} {staging_bucket} {training_steps} {learning_rate} {exported_model_versioned_uri}\n",
     "\n",
     "# Start training on CMLE.\n",
-    "gcloud ml-engine jobs submit training breast_density \\\n",
+    "gcloud ai-platform jobs submit training breast_density \\\n",
```
Comment (pavertomato) [inline imaging/ml/ml_codelab/breast_density_cloud_ml.ipynb]: this part was accidentally reverted during previous review

File: imaging/ml/ml_codelab/README.md [hunk 3]
```diff
@@ -8,6 +8,7 @@ We first need to enable the AI Platform Notebooks. Follow the steps listed [here
 
 ## Set-up permissions
 
+Go to https://cloud.google.com/healthcare/docs/resources/public-datasets/tcia#cloud-healthcare-api to request permissions to tcia dataset.
```
Comment (sadeel) [inline imaging/ml/ml_codelab/README.md]: Can you add something like "Please wait until you are granted access. You can't complete the codelab until granted access"?

File: imaging/ml/ml_codelab/breast_density_auto_ml.ipynb [hunk 4]
```diff
@@ -280,24 +138,8 @@
...
-    "\n",
-    "resp, content = http.request(path, method='POST', headers=headers, body=body)\n",
-    "assert resp.status == 200, 'error exporting to JPEG, code: {0}, response: {1}'.format(resp.status, content)\n",
-    "print('Full response:\\n{0}'.format(content.decode()))\n",
-    "\n",
-    "# Record operation_name so we can poll for it later.\n",
-    "response = json.loads(content.decode())\n",
-    "operation_name = response['name']"
+    "%%bash -s {jpeg_folder}\n",
+    "gcloud beta healthcare --project chc-tcia  dicom-stores export gcs cbis-ddsm --location=us-central1 --dataset=cbis-ddsm --mime-type=\"image/jpeg; transfer-syntax=1.2.840.10008.1.2.4.50\" --gcs-uri-prefix=$1"
```
Comment (sadeel) [inline imaging/ml/ml_codelab/breast_density_auto_ml.ipynb]: Do we need to wait until this call is completed?

**Comment by pavertomato** [comment]
> Is the part that filters to just density "2"/"3" part of this change?

as i mentioned in PR description 

> actually filtering already performed by https://github.com/GoogleCloudPlatform/healthcare/blob/master/imaging/ml/ml_codelab/scripts/tcia_utils.py#L77 and https://github.com/GoogleCloudPlatform/healthcare/blob/master/imaging/ml/ml_codelab/scripts/tcia_utils.py#L81

File: imaging/ml/ml_codelab/breast_density_auto_ml.ipynb (hunk 4 above)
Comment (pavertomato) [reply imaging/ml/ml_codelab/breast_density_auto_ml.ipynb]: yes, this action holds executions until export is done

File: imaging/ml/ml_codelab/README.md (hunk 3 above)
Comment (pavertomato) [reply imaging/ml/ml_codelab/README.md]: done

File: imaging/ml/ml_codelab/breast_density_cloud_ml.ipynb [hunk 5]
```diff
@@ -266,7 +135,7 @@
     "id": "aIfKnxqHjE9A"
    },
    "source": [
-    "Next we will convert the DICOMs to JPEGs using the ExportDicomData API. This is an asynchronous call that returns an Operation name. The operation name will be used to poll the status of the operation."
+    "Next we will convert the DICOMs to JPEGs using the ExportDicomData."
```
Comment (codezizo) [inline imaging/ml/ml_codelab/breast_density_cloud_ml.ipynb]: Can you add the link to this: https://cloud.google.com/sdk/gcloud/reference/beta/healthcare/dicom-stores/export/gcs

File: imaging/ml/ml_codelab/breast_density_auto_ml.ipynb [hunk 6]
```diff
@@ -260,16 +128,6 @@
...
    ]
   },
-  {
-   "cell_type": "markdown",
-   "metadata": {
-    "colab_type": "text",
-    "id": "aIfKnxqHjE9A"
-   },
-   "source": [
-    "Next we will convert the DICOMs to JPEGs using the ExportDicomData API. This is an asynchronous call that returns an Operation name. You can use the Operation name to poll the status of the operation."
```
Comment (codezizo) [inline imaging/ml/ml_codelab/breast_density_auto_ml.ipynb]: Not sure why the other codelab has this. If you keep this for both. Please add a link. https://cloud.google.com/sdk/gcloud/reference/beta/healthcare/dicom-stores/export/gcs

File: imaging/ml/ml_codelab/README.md [hunk 7]
```diff
@@ -8,6 +8,10 @@ We first need to enable the AI Platform Notebooks. Follow the steps listed [here
...
+
+**Please wait until you are granted access.** You can't complete the codelab until granted access.
```
[the remark this answers is not in the document]
Comment (pavertomato) [reply imaging/ml/ml_codelab/README.md]: done

File: imaging/ml/ml_codelab/README.md [hunk 8]
```diff
@@ -24,7 +28,20 @@ Follow the steps listed [here](https://cloud.google.com/ai-platform/notebooks/do
...
+```bash
+gcloud init --console-only
+```
+select `[2] Log in with a new account` and follow instructions
+
+### Run examples 
+
+In the JupyterLab UI, open File -> New Launcher, and select a Python 3 Notebook.
 
 Then run the following to import the Git repo containing the ML notebooks/code. This will clone the codelab code to your JupyterLab environment.
```
[the remark this answers is not in the document]
Comment (pavertomato) [reply imaging/ml/ml_codelab/README.md]: done

File: imaging/ml/ml_codelab/breast_density_cloud_ml.ipynb (hunk 5 above)
Comment (pavertomato) [reply imaging/ml/ml_codelab/breast_density_cloud_ml.ipynb]: done

File: imaging/ml/ml_codelab/breast_density_auto_ml.ipynb (hunk 6 above)
Comment (pavertomato) [reply imaging/ml/ml_codelab/breast_density_auto_ml.ipynb]: done
